pkg/status/conditions.go

```diff
@@ -133,11 +133,19 @@ func ConditionsForPhase(hr *v1.HelmRelease, phase v1.HelmReleasePhase) ([]v1.Hel
 		condition.Type = v1.HelmReleaseTested
 		condition.Status = v1.ConditionFalse
 		condition.Message = message
-		conditions = append(conditions, &v1.HelmReleaseCondition{
-			Type:    v1.HelmReleaseReleased,
-			Status:  v1.ConditionFalse,
-			Message: message,
-		})
+		if hr.Spec.Test.GetIgnoreFailures() {
+			conditions = append(conditions, &v1.HelmReleaseCondition{
+				Type:    v1.HelmReleaseReleased,
+				Status:  v1.ConditionTrue,
+				Message: message,
+			})
```

Comment:
I think this is redundant since the HelmReleasePhaseSucceeded should get set subsequently and handle this. Also when #452 is implemented we will need to be able to set this phase without it affecting the `Released` condition, so removing setting it here will get us closer to that.

Reply:
@seaneagan I would have thought that the 'HelmReleasePhaseSucceeded' should not be called when failed tests are ignored as you would want the helm release phase to be 'TestFailed' and not 'Succeeded'. It really depends if 'TestFailed' should be treated as a terminal state of a helm release or an interim state.

Reply:
See #404, phase is a deprecated concept. The important thing is the `Reason` that gets persisted on the conditions, which with my suggestion the `Tested` condition will be left with the `TestFailed` reason and the `Released` condition will be left in the `Succeeded` reason, when `ignoreFailures` is set and tests fail.